com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CobolWorkspaceServiceImpl.java

```diff
@@ -16,32 +16,92 @@
+import com.broadcom.lsp.domain.cobol.event.model.*;
 import com.broadcom.lsp.domain.cobol.event.model.DataEventType;
 import com.broadcom.lsp.domain.cobol.event.model.RequiredCopybookEvent;
 import com.broadcom.lsp.domain.cobol.event.model.RunAnalysisEvent;
 import com.google.inject.Inject;
+import com.google.inject.Provider;
 import com.google.inject.Singleton;
 import lombok.extern.slf4j.Slf4j;
 import org.eclipse.lsp4j.DidChangeConfigurationParams;
 import org.eclipse.lsp4j.DidChangeWatchedFilesParams;
+import org.eclipse.lsp4j.*;
+import org.eclipse.lsp4j.services.LanguageClient;
 
 import javax.annotation.Nonnull;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.FileVisitOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Stream;
 
 @Slf4j
 @Singleton
 public class CobolWorkspaceServiceImpl implements CobolWorkspaceService {
   private final DefaultDataBusBroker dataBus;
+  private Provider<LanguageClient> clientProvider;
 
   @Inject
-  public CobolWorkspaceServiceImpl(DefaultDataBusBroker dataBus) {
+  public CobolWorkspaceServiceImpl(
+      DefaultDataBusBroker dataBus, Provider<LanguageClient> clientProvider) {
     this.dataBus = dataBus;
+    this.clientProvider = clientProvider;
     dataBus.subscribe(DataEventType.REQUIRED_COPYBOOK_EVENT, this);
   }
 
+  /**
+   * This is a notification triggered automatically when the user modify configuration settings in
+   * the client
+   *
+   * @param params - LSPSpecification -> The actual changed settings; Actually -> null all the time.
+   */
   @Override
   public void didChangeConfiguration(DidChangeConfigurationParams params) {
-    throw new UnsupportedOperationException("Functionality not implemented yet");
+    /** section and scope has to be set to whatever we agree on for the dependencies graph */
+    try {
+      fetchSettings(null, null)
+          .thenAccept(
+              e -> {
+                dataBus.postData(FetchedSettingsEvent.builder().content(e).build());
+              });
+    } catch (Exception e) {
+      log.error(e.getMessage());
+    }
+  }
+
+  /**
+   * After client notifies the server that there is a setting change we need to request the client
+   * those changing by sending a workspace/configuration JSON request
+   *
+   * @param section - The configuration section asked for.
+   * @param scope - The scope to get the configuration section for.
+   * @return - CompletedFuture which contains an object with the settings asked for.
+   */
+  private CompletableFuture<List<Object>> fetchSettings(String section, String scope) {
+    LanguageClient client = clientProvider.get();
+    ConfigurationParams params = new ConfigurationParams();
+    List<ConfigurationItem> itemList = new ArrayList<>();
+    elemToList(itemList, section, scope);
```

Comment:
I want also say that also pass null as function argument is a bad practice until is not strictly required (and in case should be explained the reason)